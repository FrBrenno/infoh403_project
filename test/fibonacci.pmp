'' This prompt is a piece of code that should work ''
'' As there is no string definition in our language, we will use comments instead. ''

begin
    ** Prompt the user for the number of terms
    print(a)... **Imagine that a is a variable containing the string "Fibonnaci Terms"

    numTerms := 5...

    ** Initialize variables
    term1 := 0...
    term2 := 1...
    i := 1...

    ** Display the first two terms
    print(term1)...
    print(term2)...

    ** Calculate and display the remaining terms
    while i < numTerms - 2 do
        begin
            nextTerm := term1 + term2...
            print(nextTerm)...
            term1 := term2...
            term2 := nextTerm...
            i := i + 1
        end
end

