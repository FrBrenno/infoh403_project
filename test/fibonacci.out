token: begin	lexical unit: BEG
token: print	lexical unit: PRINT
token: (	lexical unit: LPAREN
token: )	lexical unit: RPAREN
token: numTerms	lexical unit: VARNAME
token: :=	lexical unit: ASSIGN
token: 5	lexical unit: NUMBER
token: term1	lexical unit: VARNAME
token: :=	lexical unit: ASSIGN
token: 0	lexical unit: NUMBER
token: term2	lexical unit: VARNAME
token: :=	lexical unit: ASSIGN
token: 1	lexical unit: NUMBER
token: i	lexical unit: VARNAME
token: :=	lexical unit: ASSIGN
token: 1	lexical unit: NUMBER
token: print	lexical unit: PRINT
token: (	lexical unit: LPAREN
token: term1	lexical unit: VARNAME
token: )	lexical unit: RPAREN
token: print	lexical unit: PRINT
token: (	lexical unit: LPAREN
token: term2	lexical unit: VARNAME
token: )	lexical unit: RPAREN
token: while	lexical unit: WHILE
token: {	lexical unit: LBRACK
token: i	lexical unit: VARNAME
token: <	lexical unit: SMALLER
token: =	lexical unit: EQUAL
token: numTerms	lexical unit: VARNAME
token: -	lexical unit: MINUS
token: 2	lexical unit: NUMBER
token: }	lexical unit: RBRACK
token: do	lexical unit: DO
token: begin	lexical unit: BEG
token: nextTerm	lexical unit: VARNAME
token: :=	lexical unit: ASSIGN
token: term1	lexical unit: VARNAME
token: +	lexical unit: PLUS
token: term2	lexical unit: VARNAME
token: print	lexical unit: PRINT
token: (	lexical unit: LPAREN
token: nextTerm	lexical unit: VARNAME
token: )	lexical unit: RPAREN
token: term1	lexical unit: VARNAME
token: :=	lexical unit: ASSIGN
token: term2	lexical unit: VARNAME
token: term2	lexical unit: VARNAME
token: :=	lexical unit: ASSIGN
token: nextTerm	lexical unit: VARNAME
token: i	lexical unit: VARNAME
token: :=	lexical unit: ASSIGN
token: i	lexical unit: VARNAME
token: +	lexical unit: PLUS
token: 1	lexical unit: NUMBER
token: end	lexical unit: END
token: end	lexical unit: END

Variables
term2 11
term1 10
numTerms 7
i 12
nextTerm 21
